### PR TITLE
crypto: add test case for seeding RNG pool

### PR DESCRIPTION
**Pull request by igoropaniuk**
Add test case for testing system pTA functionality for seeding RNG pool.

Relates to: https://github.com/OP-TEE/optee_os/pull/2340

`Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>`

### CODE REVIEW

**Comment by igoropaniuk** [comment]
@jbech-linaro all comments addressed

**Comment by jbech-linaro** [comment]
LGTM!
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by igoropaniuk** [comment]
@jbech-linaro tag applied, rebased and ready for merge. Thanks!

**Comment by igoropaniuk** [comment]
> It looks like the test will fail when CFG_SYSTEM_PTA=n, no? That should not happen.

Right, thanks! Fixed
Also check please another part of this fix in optee-os@2340 (https://github.com/OP-TEE/optee_os/pull/2340/commits/9c47e15a9d31d7315ec8af3c64486e04470c3ecf)

File: host/xtest/regression_4000.c [hunk 1]
```diff
@@ -41,7 +41,10 @@ static void xtest_tee_test_4008(ADBG_Case_t *Case_p);
 static void xtest_tee_test_4009(ADBG_Case_t *Case_p);
 static void xtest_tee_test_4010(ADBG_Case_t *Case_p);
 static void xtest_tee_test_4011(ADBG_Case_t *Case_p);
+
```
Comment (jforissier) [inline host/xtest/regression_4000.c]: Blank line not needed

File: host/xtest/regression_4000.c [hunk 2]
```diff
@@ -65,8 +68,11 @@ ADBG_CASE_DEFINE(regression, 4010, xtest_tee_test_4010,
 		"Test TEE Internal API create transient object (negative)");
 ADBG_CASE_DEFINE(regression, 4011, xtest_tee_test_4011,
 		"Test TEE Internal API Bleichenbacher attack (negative)");
+
```
Comment (jforissier) [inline host/xtest/regression_4000.c]: Blank line not needed

File: ta/crypt/ta_entry.c [hunk 3]
```diff
@@ -32,7 +32,10 @@
 #include <ta_crypt.h>
 #include <tee_ta_api.h>
 #include <trace.h>
+
+#ifdef CFG_SYSTEM_PTA
```
Comment (jforissier) [inline ta/crypt/ta_entry.c]: `#ifdef` not needed + alphabetical order please

**Comment by igoropaniuk** [comment]
@jforissier 
Fixed and tag applied, thanks!

**Comment by igoropaniuk** [comment]
Rebased on the latest master

**Pull request merged**